public/app/features/alerting/components/AlertingQueryRows.tsx

```diff
@@ -30,14 +23,15 @@ interface State {
 export class AlertingQueryRows extends PureComponent<Props, State> {
   constructor(props: Props) {
     super(props);
+
     this.state = { dataPerQuery: {} };
   }
 
   onRemoveQuery = (query: DataQuery) => {
     this.props.onQueriesChange(this.props.queries.filter((item) => item.model !== query));
   };
 
-  onChangeTimeRange(timeRange: RelativeTimeRange, index: number) {
+  onChangeTimeRange = (timeRange: RelativeTimeRange, index: number) => {
```

Comment:
Issues with `this` bindings or why the arrow functions?

Reply:
yes, since we are passing it to the `AlertingWrapper` component.